enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?
+    to_a.my_each do |i|
+      return false if yield(i) == false
+      else
+        return true
+      end
+    end
+  end
```

Comment:
***my_all?***

I will be using these variables for the next suggestions:

```
true_array = [1, true, 'hi', []]
array = [2, 5, 9, 7]
```

- [ ] When no block or argument is given returns true when none of the collection members are false or nil:

```
     true_array.my_all?
    
     LocalJumpError:
       no block given (yield)
     # ./enumerables.rb:34:in `block in my_all?'
     # ./enumerables.rb:7:in `my_each'
     # ./enumerables.rb:33:in `my_all?'
```

- [ ]  When a class is passed as an argument returns true if all of the collection is a member of such class::Integer (Your method should be able to take an argument, such as a Regex, a Class, or a number):

```
     array.my_all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     # ./enumerables.rb:32:in `my_all?'
```